hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -539,12 +541,22 @@ Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
             database.setDescription(value);
           } else if (key.equals("location")) {
             database.setLocationUri(value);
+          } else if (key.equals(TableProperties.HMS_DB_OWNER)) {
+            database.setOwnerName(value);
+          } else if (key.equals(TableProperties.HMS_DB_OWNER_TYPE) && value != null) {
+            database.setOwnerType(PrincipalType.valueOf(value));
```

Comment:
I believe that valueOf() can throw an IllegalArgumentException if there is no such enum value as the input parameter. This exception might be handled on the API level but we might want to take care of it more gracefully here.
Additionally, some test coverage for this would be great.

Reply:
yes, nice catch.

Reply:
see the precondition check added above

Reply:
See my comment above about preconditions.